mutters-core/src/main/java/com/rabidgremlin/mutters/core/CompoundIntentMatcher.java

```diff
@@ -22,32 +24,32 @@
   /**
    * Constructor for the class.
    * 
-   * @param firstMatcher The first intent matcher.
+   * @param firstMatcher  The first intent matcher.
    * @param secondMatcher The second intent matcher.
    */
   public CompoundIntentMatcher(IntentMatcher firstMatcher, IntentMatcher secondMatcher)
   {
-    this.firstMatcher = firstMatcher;
-    this.secondMatcher = secondMatcher;
+    this.firstMatcher = Objects.requireNonNull(firstMatcher);
+    this.secondMatcher = Objects.requireNonNull(secondMatcher);
   }
 
   /*
    * (non-Javadoc)
    * 
-   * @see com.rabidgremlin.mutters.core.IntentMatcher#match(String utterance, Context context, Set<String>
-   * expectedIntents)
+   * @see com.rabidgremlin.mutters.core.IntentMatcher#match(String utterance,
+   * Context context, Set<String> expectedIntents)
    * 
    */
   @Override
-  public IntentMatch match(String utterance, Context context, Set<String> expectedIntents, HashMap<String, Object> debugValues)
+  public IntentMatch match(String utterance, Context context, Set<String> expectedIntents)
   {
     // see if we can find match in first matcher
-    IntentMatch match = firstMatcher.match(utterance, context, expectedIntents, debugValues);
+    IntentMatch match = firstMatcher.match(utterance, context, expectedIntents);
 
     // no ? try second one
-    if (match == null)
+    if (match.matched())
```

Comment:
this needs to be `!match.matched()`

Reply:
Oops... guess I don't have any tests covering this one!

Reply:
Fixed and tests added.